### DISCUSSION TITLE
Reuse media table

### DISCUSSION THREAD

**Discussion by ArtMin96** [awcodes/filament-curator#427] [Q&A]
If I use multiple, should I create a `some_model_media` pivot table for each model that should have multiple images?

**Comment by awcodes** [comment]
Yea. That's standard laravel relationships. If you have multiple models that need to have multiple images then you will need a pivot table.

**Reply by ArtMin96** [reply]
Ok. Thanks! It would be great if it was polymorphic

**Reply by awcodes** [reply]
If the media table itself was polymorphic you wouldn't be able to reuse the same media on different models.

**Reply by ArtMin96** [reply]
By reuse, I mean the table, not the image. For example: `mediable_type`, `mediable_id`

**Reply by awcodes** [reply]
That's my point. In the real world a media manager means that a blog post and a page could use the same "hero" image. If the media table was polymorphic by default then they both couldn't use the same image. It's one of the main reasons I don't support spatie's media library with this plugin.
